tests/components/telegram_bot/test_telegram_bot.py

```diff
@@ -108,27 +95,6 @@ async def test_webhook_endpoint_generates_telegram_callback_event(
     assert events[0].data["data"] == update_callback_query["callback_query"]["data"]
 
 
-async def test_polling_platform_message_text_update(
-    hass: HomeAssistant, polling_platform, update_message_text
-) -> None:
-    """Provide the `PollBot`s `Dispatcher` with an `Update` and assert fired `telegram_text` event."""
-    events = async_capture_events(hass, "telegram_text")
-
-    def telegram_dispatcher_callback():
-        dispatcher = Dispatcher.get_instance()
-        update = Update.de_json(update_message_text, dispatcher.bot)
-        dispatcher.process_update(update)
-
-    # python-telegram-bots `Updater` uses threading, so we need to schedule its callback in a sync context.
-    await hass.async_add_executor_job(telegram_dispatcher_callback)
```

Comment:
I think we can patch like this to get access to the callback which we want to call here.

```py
with patch("homeassistant.components.telegram_bot.polling.ApplicationBuilder") as application_builder_class:
    application = application_builder_class.return_value.bot.return_value.build.return_value
    handler = application.add_handler.call_args[0][0]
    callback = handler.callback

# Set up the integration with the polling platform inside the patch context manager. Then call the callback and assert events fired.
```

Reply:
🙏 appreciate your direction on that